includes/class-wc-stripe-helper.php

```diff
@@ -337,6 +337,117 @@ public static function get_settings( $method = null, $setting = null ) {
 		return isset( $all_settings[ $setting ] ) ? $all_settings[ $setting ] : '';
 	}
 
+	/**
+	 * List of legacy payment methods.
+	 *
+	 * @return array
+	 */
+	public static function get_legacy_payment_methods() {
+		$payment_method_classes = [
+			WC_Gateway_Stripe_Bancontact::class,
+			WC_Gateway_Stripe_EPS::class,
+			WC_Gateway_Stripe_Giropay::class,
+			WC_Gateway_Stripe_Ideal::class,
+			WC_Gateway_Stripe_p24::class,
+			WC_Gateway_Stripe_Sepa::class,
+			WC_Gateway_Stripe_Boleto::class,
+			WC_Gateway_Stripe_Oxxo::class,
+		];
+
+		/** Show Sofort if it's already enabled. Hide from the new merchants and keep it for the old ones who are already using this gateway, until we remove it completely.
+		 * Stripe is deprecating Sofort https://support.stripe.com/questions/sofort-is-being-deprecated-as-a-standalone-payment-method.
+		 */
+		$sofort_settings = get_option( 'woocommerce_stripe_sofort_settings', [] );
+		if ( isset( $sofort_settings['enabled'] ) && 'yes' === $sofort_settings['enabled'] ) {
+			$payment_method_classes[] = WC_Gateway_Stripe_Sofort::class;
+		}
+
+		$payment_methods = [];
+
+		foreach ( $payment_method_classes as $payment_method_class ) {
+			$payment_method                         = new $payment_method_class();
+			$payment_methods[ $payment_method->id ] = $payment_method;
+		}
+
+		return $payment_methods;
+	}
+
+	/**
+	 * List of available legacy payment methods.
+	 *
+	 * @return array
+	 */
+	public static function get_legacy_available_payment_method_ids() {
+		$payment_methods = self::get_legacy_payment_methods();
+
+		// In legacy mode (when UPE is disabled), Stripe refers to card as payment method.
+		$available_payment_method_ids = [ 'card' ];
+
+		foreach ( $payment_methods as $payment_method ) {
+			$payment_method_id              = 'stripe_sepa' === $payment_method->id ? 'sepa_debit' : str_replace( 'stripe_', '', $payment_method->id );
+			$available_payment_method_ids[] = $payment_method_id;
+		}
+
+		return $available_payment_method_ids;
+	}
+
+	/**
+	 * List of enabled legacy payment methods.
+	 *
+	 * @return array
+	 */
+	public static function get_legacy_enabled_payment_methods( $field = null ) {
```

Comment:
Since the only field we retrieve is "id", would it be worth to reuse `get_legacy_available_payment_method_ids()` for this? The logic seems similar, and we could use an argument in that method instead of in this one to only return the enabled gateways.